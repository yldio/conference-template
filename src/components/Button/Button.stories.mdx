import {Meta, Story, Preview, Props, Description} from '@storybook/addon-docs/blocks'
import {text, addDecorator} from '@storybook/addon-knobs/react'
import { action } from '@storybook/addon-actions';
import Button, {Variants} from './'
import {ThemeProvider} from 'theme-ui'
import defaultTheme from '../../themes/default'

<Meta 
  title="Atoms|Button" 
  component={Button} 
  decorators={[(storyFn) => <ThemeProvider theme={defaultTheme}>{storyFn()}</ThemeProvider>]}
/>

# Button

The button abstract a HTML `<button>` element, with styles according to the
theme being used.  It's ideal to be used in actions, like forms, or part of a
UI interaction.

> Don't use the `Button` component to represent links to other pages.

<Preview withToolbar>
  <Story name="default">
      <Button onClick={action('onClick')}>{text('children', 'Hello World')}</Button>
  </Story>
</Preview>

<Props of={Button} />

## Variants

<Preview withToolbar> 
  <Story name="Dark Variant">
    <Button variant={Variants.Dark}>{text('children', 'Hello World')}</Button>
  </Story>
  <Story name="Light Variant"> 
    <Button variant={Variants.Light}>{text('children', 'Hello World')}</Button>
  </Story>
</Preview>


