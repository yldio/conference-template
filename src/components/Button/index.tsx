/** @jsx jsx */

import React from 'react';
import { jsx } from 'theme-ui';

interface Props {
  /**
   * There two variants for the default theme
   */
  variant?: 'light' | 'dark';
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
  children: React.ReactNode;
}

const Button: React.SFC<Props> = ({ children, variant, onClick }) => (
  <button sx={{ variant: `buttons.${variant}` }} onClick={onClick}>
    {children}
  </button>
);

Button.defaultProps = {
  variant: 'dark',
};

export default Button;
