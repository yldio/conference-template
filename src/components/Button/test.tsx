import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button, { Variants } from './';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom/extend-expect';

describe.only('Button', () => {
  it('snapshot', () => {
    const { container } = render(
      <Button variant={Variants.Dark}>Hello World</Button>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the given text', () => {
    const text = 'Hello World';
    const { getByText } = render(<Button>{text}</Button>);
    const content = getByText(text);

    expect(content).toBeInTheDocument();
  });

  it('should call onClick event handler', () => {
    const text = 'Hello World';
    const onClick = jest.fn();
    const { getByText } = render(<Button onClick={onClick}>{text}</Button>);

    userEvent.click(getByText(text));

    expect(onClick).toHaveBeenCalled();
  });
});
