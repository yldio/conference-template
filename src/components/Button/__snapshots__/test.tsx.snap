// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<button
  class="css-0"
>
  Hello World
</button>
`;
